lib/developer_portal/config/routes.rb

```diff
@@ -45,10 +45,14 @@
         end
       end
 
-      resource :braintree_blue, :stripe, only: [:show, :edit] do
+      resource :braintree_blue, only: [:show, :edit] do
         match 'hosted_success', via: [:get, :post], on: :member
       end
 
+      resource :stripe, only: %i[show edit update] do
```

Comment:
Adding `update` method for Stripe

Reply:
I'm a bit lost on the workflow for this.

- Why don't authorize.net and braintree have an `:update` route?
- Why is there an `update` method in `PaymentDetailsBaseController`, if there was no route for the `:update` action for braintree and authorize.net?
- What is the role of `hosted_success`?

Reply:
I'll try to explain (although there are things that I am not sure about either):

- This is related to this change: https://github.com/3scale/porta/pull/3696/files#diff-68ed5ea9c56938c39a89763a6b9d9264828b50b404de472c78d6c455bc6faffaR11-R16
Previously the `update` method was handled by the generic `PaymentDetailsBaseController` for all gateways. In order to distinguish Stripe from other gateways (because only Stripe needs to update the billing address in the same flow), I am forcing the "update details" form to go to the StripeController and not the generic one.